Lib/test/test_dataclasses.py

```diff
@@ -2890,6 +2903,35 @@ class S(D):
         self.assertEqual(s.y, 10)
         self.assertEqual(s.cached, True)
 
+        with self.assertRaises(FrozenInstanceError):
+            del s.x
+        self.assertEqual(s.x, 3)
+        with self.assertRaises(FrozenInstanceError):
+            del s.y
+        self.assertEqual(s.y, 10)
+        del s.cached
+        self.assertFalse(hasattr(s, 'cached'))
+        with self.assertRaises(AttributeError):
+            del s.cached
```

Comment:
Should we also check the raised exception is exactly `AttributeError` rather than `FrozenInstanceError` (subclass of `AttributeError`)? This exception is raised from `super().__delattr__(...)`.

Reply:
That would get you extra brownie points :)

You could do it like this:

```suggestion
        with self.assertRaises(AttributeError) as cm:
            del s.cached
        
        self.assertNotIsInstance(cm.exception, FrozenInstanceError)
```

Reply:
Thanks for the guidance. Added.